shell.nix

```diff
@@ -1,5 +1,7 @@
 { system ? builtins.currentSystem
-, config ? {}
+, config ? { allowUnfreePredicate = (import ./lib.nix {}).unfreePredicate;
```

Comment:
Ouch. This line is actually kind of weird. I'd just drop the system and config arguments so the import uses the defaults.

Reply:
I kept this in because I needed them at the time, I will check if I still need them but if I drop them completely then since it is `shell.nix` it will use the system defaults will it not, rather than the lib versions (that are in default.nix)?

Reply:
I mean, I think you don't need to have `system` and `config` arguments at all, and then just don't pass anything to `default.nix` when you import it. Seems to work for me but maybe there's something wrong with it?

Reply:
ok, I'll check it once I've managed to get the buildkite thing fixed (could be all year)

Reply:
yup, all good, tested on OSX and NixOS, thanks.